Automattic/Gravatar-SDK-iOS
# GravatarUI docc bundle

Closes #155 

### Description

This PR implements the basics to add docs and pages to GravatarUI using Swfit-docc.

I've added a new snapshot test case to generate an example profile view to use in the docs. For now I've decided to use the same account that is used in https://docs.gravatar.com/general/profile/ , though this this is subject to change. More important than the account itself, is to check the strategy chosen to generate and use these snapshots.

I've created a make command to copy the snapshots from the test target to the docs bundle:
`make update-example-snapshots`

This is not hook to anything yet, so it need to be triggered manually.
Sadly in SPM we don't have a way to trigger post-test commands. 
We can add a post-action to the test target in the Xcode scheme, but it's a local setting.

### Testing Steps

- Checkout the banch
- Build the project
- On Xcode, go to Product -> Build Documentation
- Though the content is also subject to change, is enough to see that the main GravatarUI page renders correctly.

![CleanShot 2024-04-24 at 18 33 32@2x](https://github.com/Automattic/Gravatar-SDK-iOS/assets/9772967/24c25784-a5ec-46e9-afe1-319c1b9a61a2)
